src/Microsoft.AspNet.Razor/Parser/TagHelpers/TagHelperBlockRewriter.cs

```diff
@@ -28,7 +28,11 @@ public static class TagHelperBlockRewriter
             var attributes = GetTagAttributes(tagName, validStructure, tag, descriptors, errorSink);
             var selfClosing = IsSelfClosing(tag);
 
-            return new TagHelperBlockBuilder(tagName, selfClosing, start, attributes, descriptors);
+            // All descriptors for an element should share a common prefix, grab the first.
+            var prefix = descriptors.First().Prefix;
+            var tagHelperTagName = tagName.Substring(prefix.Length);
```

Comment:
Actually scratch that, need this for testing scenarios and also in the `TagHelperParseTreeRewriter` matching scenario, we can't just pull from the `TagHelperDescriptor`s because their tagname may be `*`, just realized the `*` case is a hole I need to fix.